cargo-shuttle/src/args.rs

```diff
@@ -228,78 +226,130 @@ pub struct RunArgs {
+pub struct TemplateLocation {
+    pub auto_path: String,
+    pub subfolder: Option<String>,
+}
+
 impl InitArgs {
-    /// `None` -> No template chosen, ask for it
-    ///
-    /// `Some(Template::None)` -> Init with a blank cargo project
-    pub fn framework(&self) -> Option<Template> {
-        // Why separate enums?
-        // Future might have more options that pre-defined templates
-        self.template.as_ref().map(|t| {
-            use InitTemplateArg::*;
-            match t {
-                ActixWeb => Template::ActixWeb,
-                Axum => Template::Axum,
-                Poem => Template::Poem,
-                Poise => Template::Poise,
-                Rocket => Template::Rocket,
-                Salvo => Template::Salvo,
-                Serenity => Template::Serenity,
-                Tide => Template::Tide,
-                Thruster => Template::Thruster,
-                Tower => Template::Tower,
-                Warp => Template::Warp,
-                None => Template::None,
-            }
-        })
+    pub fn git_templates(&self) -> Option<Vec<TemplateLocation>> {
+        if let Some(from) = self.from.clone() {
+            Some(vec![TemplateLocation {
+                auto_path: from,
+                subfolder: self.subfolder.clone(),
+            }])
+        } else {
+            self.template.as_ref().map(|t| t.templates())
+        }
+    }
+}
+
+impl InitTemplateArg {
+    pub fn templates(&self) -> Vec<TemplateLocation> {
+        use InitTemplateArg::*;
+        let paths = match self {
+            // first entry should be the default for
+            // that choice of framework (usually hello-world)
+            ActixWeb => vec![
+                "actix-web/hello-world",
+                "actix-web/postgres",
+                "actix-web/websocket-actorless",
+            ],
+            Axum => vec![
+                "axum/hello-world",
+                "axum/static-files",
+                "axum/static-next-server",
+                "axum/websocket",
+                "axum/with-state",
+            ],
+            Poem => vec!["poem/hello-world", "poem/mongodb", "poem/postgres"],
+            Poise => vec!["poise/hello-world"],
+            Rocket => vec![
+                "rocket/hello-world",
+                "rocket/authentication",
+                "rocket/dyn_template_hbs",
+                "rocket/persist",
+                "rocket/postgres",
+                "rocket/secrets",
+                "rocket/url-shortener",
+                "rocket/workspace",
+            ],
+            Salvo => vec!["salvo/hello-world"],
+            Serenity => vec!["serenity/hello-world", "serenity/postgres"],
+            Thruster => vec!["thruster/hello-world", "thruster/postgres"],
+            Tide => vec!["tide/hello-world", "tide/postgres"],
+            Tower => vec!["tower/hello-world"],
+            Warp => vec!["warp/hello-world"],
+            None => vec!["custom/none"],
```

Comment:
1. I included all examples that are related to the listed frameworks. But yeah, some rule to follow would be good, I'd use this as a first version though.
2. The link is shown, see screenshot at top. After init is cloning, the examples readme will be updated with proper instructions, and perhaps a growing list of community templates.

Reply:
The value is that users who are unfamiliar with Shuttle or the examples repo get a better view of what is being offered from the official examples. If they follow the link they can learn more about which templates are offered, but also how to clone any template and check out the community list.
You also wont need to remember/copy+paste the name of common templates that you use ("Let me make another axum app with postgres" -> easy peasy, no lookup of repo name and path).

Reply:
> If they follow the link they can learn more about which templates are offered, but also how to clone any template and check out the community list.

This is sufficient.

> You also wont need to remember/copy+paste the name of common templates that you use ("Let me make another axum app with postgres" -> easy peasy, no lookup of repo name and path).

I see the point, but at the same time, it can be an `init --from <github_repo_url>` which would make it much faster than going through the entire init flow. 

> . If they follow the link they can learn more about which templates are offered, but also how to clone any template and check out the community list.
We already tell them how to do it when initing (you show up that HINT). I would rather defend against introducing uncertainty than add learning material in the CLI (makes me think I am doing a lab).

Also, what about initing a template with an older cargo-shuttle than the deps from the repo?
Also, why showing up templates for multiple frameworks with `postgres` (for example)? Users might think each framework has different requirements. If we add only one then, they'll whether the resource is supported only for a framework? After which point they'll look out to the docs :) I would avoid all these qs and any future ones related to which other templates to add to the CLI (I guarantee that if we add an sqlite one, we would one a template for it, and the same for dynamo-db, and then for s3, and the list goes one until you have a build config system similar to the Linux kernel. 

Let's keep it simple stupid from an init perspective and rely on the docs for any learning aspect. The DevX is very important for us, but I am not sure though.